tracer/src/Datadog.Trace.Tools.dd_dotnet/Checks/ProcessConfiguration.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="ProcessConfiguration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Datadog.Trace.Tools.Shared;
+
+namespace Datadog.Trace.Tools.dd_dotnet.Checks;
+
+internal static class ProcessConfiguration
+{
+    internal static IConfigurationSource ExtractConfigurationSource(this ProcessInfo process,  string? baseDirectory, IReadOnlyDictionary<string, string>? appSettings)
+    {
+        baseDirectory ??= Path.GetDirectoryName(process.MainModule);
+
+        var configurationSource = new CompositeConfigurationSource();
```

Comment:
Don't you want to be using `CompositeConfigurationSourceInternal` here? 🤔 Because this is a public API, and will try to record telemetry for it

Reply:
That's the dd-dotnet CompositeConfigurationSource, not the tracer one. It has no telemetry. I can rename it if it's too confusing, but I don't have great ideas for the new name.